iceoryx_hoofs/include/iceoryx_hoofs/internal/relocatable_pointer/pointer_repository.hpp

```diff
@@ -45,33 +46,40 @@ class PointerRepository
         ptr_t endPtr{nullptr};
     };
 
-    /// @note 0 is a special purpose id and reserved
-    /// id 0 is reserved to interpret the offset just as a raw pointer,
-    /// i.e. its corresponding base ptr is 0
     static constexpr id_t MIN_ID{1U};
     static constexpr id_t MAX_ID{CAPACITY - 1U};
 
     static_assert(MAX_ID >= MIN_ID, "MAX_ID must be greater or equal than MIN_ID!");
     static_assert(CAPACITY >= 2, "CAPACITY must be at least 2!");
 
   public:
-    static constexpr id_t INVALID_ID = std::numeric_limits<id_t>::max();
+    /// @note 0 is a special purpose id and reserved
+    /// id 0 is reserved to interpret the offset just as a raw pointer,
+    /// i.e. its corresponding base ptr is 0
+    static constexpr id_t RAW_POINTER_BEHAVIOUR{0};
 
     /// @brief default constructor
     PointerRepository() noexcept;
+    ~PointerRepository() noexcept = default;
+
+    PointerRepository(const PointerRepository&) = delete;
+    PointerRepository(PointerRepository&&) = delete;
+    PointerRepository& operator=(const PointerRepository&) = delete;
+    PointerRepository& operator=(PointerRepository&&) = delete;
 
     /// @brief registers the start pointer of the segment in another application with a specific id
-    /// @param[in] id identifies the segment
+    /// @param[in] id identifies the segment which the pointer should be added to
     /// @param[in] ptr is the start pointer of the segment
     /// @param[in] size is the size of the segment
     /// @return true if the registration was successful, otherwise false
-    bool registerPtr(id_t id, ptr_t ptr, uint64_t size) noexcept;
+    bool registerPtrWithId(id_t id, ptr_t ptr, uint64_t size) noexcept;
```

Comment:
I prefer overloading to long names. This is how it was done in C in the past (and had to, because overloading did not exist).

Reply:
@MatthiasKillat In general I agree with you, but in this case the parameters are rather similar so it's nicer if you type `.` in your IDE and then get two different methods offered.